ci-operator/step-registry/hypershift/mce/power/create/ibmcloud-workers/boot/hypershift-mce-power-create-ibmcloud-workers-boot-commands.sh

```diff
@@ -0,0 +1,265 @@
+
+	# Extracting iso download url
+	DISCOVERY_ISO_URL=$(oc get infraenv/${INFRAENV_NAME} -n ${HOSTED_CONTROL_PLANE_NAMESPACE} -o json | jq -r '.status.isoDownloadURL')
+	DISCOVERY_ISO_DOWNLOAD_LINK_FILE="/tmp/${HOSTED_CLUSTER_NAME}-iso-download-link"
+	echo ${DISCOVERY_ISO_URL} >${DISCOVERY_ISO_DOWNLOAD_LINK_FILE}
+
+	# Create private key with 0600 permission for ssh purpose
+	SSH_PRIVATE="/tmp/ssh-privatekey"
+	cp "${AGENT_POWER_CREDENTIALS}/ssh-privatekey" ${SSH_PRIVATE}
+	chmod 0600 ${SSH_PRIVATE}
+
+	SSH_OPTIONS=(-o 'PreferredAuthentications=publickey' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -i "${SSH_PRIVATE}")
+
+	# Scp download link file
+	echo "$(date) Scp iso download link"
+	scp "${SSH_OPTIONS[@]}" ${DISCOVERY_ISO_DOWNLOAD_LINK_FILE} root@${BASTION}:${DISCOVERY_ISO_DOWNLOAD_LINK_FILE}
+
+	IFS=' ' read -r -a HOST_NAMES <<<"${INSTANCE_NAMES}"
+
+	serverArgs=""
+	for ((i = 0; i < ${HYPERSHIFT_NODE_COUNT}; i++)); do
+		serverArgs+="${HOST_NAMES[i]},${MAC_ADDRESSES[i]},${IP_ADDRESSES[i]} "
+		echo $serverArgs
+	done
+
+	# Setup pxe boot on bastion for agents to boot
+	echo "$(date) Setup pxe boot in bastion"
+	ssh "${SSH_OPTIONS[@]}" root@${BASTION} "cd ${BASTION_CI_SCRIPTS_DIR} && ./setup-pxe-boot.sh ${HOSTED_CLUSTER_NAME} ${HYPERSHIFT_NODE_COUNT} ${serverArgs}"
+
+	sleep 240
+
+	# Rebooting vm to boot from the network
+	for instance in "${HOST_NAMES[@]}"; do
+		ibmcloud pi ins act $instance -o soft-reboot
+	done
+}
+
+boot_x86_vm() {
+	# shellcheck disable=SC2034
+	WORKER_IP=$1
+	# Log into workers via Bastion
+
+	# shellcheck disable=SC2087
+	ssh "${SSH_OPTIONS[@]}" root@${BASTION} <<EOF
+    cd /var/www/html && curl -k -L -o rootfs.img "${ROOTFS_URL}"
+    timeout 120 ssh -o 'StrictHostKeyChecking=no' -i ${BASTION_KEY} root@${WORKER_IP} << WEOF
+    echo "Logged into the worker.."
+    cd /var && mkdir -p home && cd /var/home && mkdir core && cd /var/home/core
+    touch boot.sh
+    cat <<FEOF > "boot.sh"
+    #!/bin/bash
+    set -x 
+
+    curl -k -L -o /var/home/core/initrd.img "${INITRD_URL}"
+    curl -k -L -o /var/home/core/kernel.img "${KERNEL_URL}"
+    sudo kexec -l /var/home/core/kernel.img --initrd=/var/home/core/initrd.img --append="rd.neednet=1 coreos.live.rootfs_url=http://${BASTION}:443/rootfs.img random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal console=tty1 console=ttyS1,115200n8"
+    sudo kexec -e &
+    echo "Initiated agent bootup"
+FEOF
+    cat boot.sh
+    chmod +x boot.sh
+    ./boot.sh
+    echo "$(date) Running agent bootup"
+WEOF
+echo "$(date) Triggered worker reboot"
+EOF
+	echo "$(date) Ran agent bootup"
+}
+
+boot_x86_workers() {
+	IP_X86=$(cat ${SHARED_DIR}/ipx86addr)
+	# shellcheck disable=SC2034
+	BASTION_KEY=/root/agent-key
+
+	# shellcheck disable=SC2034
+	INITRD_URL=$(oc get infraenv ${HOSTED_CLUSTER_NAME}-x86 -n ${HOSTED_CONTROL_PLANE_NAMESPACE} -o json | jq -r '."status"."bootArtifacts"."initrd"')
```

Comment:
Any specific reason for this?

Reply:
Yes, once you cleanup a HCP, that Namespace will be cleaned up too and both Infraenv and HCP Namespace with the rest of the objects are different from each other, so they should keep different life-cycle changes.

One example is:

- I want to create 2 HostedClusters with a number of workers that could move from the first HostedCluster to the second at some point. My Infraenv will keep 3 BM nodes
- First I provision a HC with 2 nodes, this 2 nodes belongs to the Infraenv (They could be assigned to a HC or not)
- Once provision is finished, the Nodes are assigned to the first HC.
- Now I de-provision the HC, the nodes will be moved to the Infraenv appearing as new agents (loaded with the discovery iso) and waiting for the new HC to be assigned to.
- I decided to provision another cluster, with other 2 nodes.
- The new cluster will hae only assigned 2/3 nodes, so we can scale up

As you can see in this example, if we keep the same namespace for Infraenv and for HCP this will make the Infraenv to be deleted after the first HC deletion.